nbcparser/main.go

```diff
@@ -1,40 +1,55 @@
 package main
 
 import (
+	"flag"
 	"log"
 	"os"
+	"os/exec"
 
 	"github.com/Azure/agentbaker/nbcparser/pkg/parser"
 )
 
-// input to this function will be the serialized JSON from userdata + custom data.
-// it will be deserialized to the contract that the VHD this binary will be on supports.
-// Parse will be called using that deserialized struct and output the generated cse_cmd to trigger the bootstrap process.
-// example usage:
-// to build: go build main.go.
-// to run: ./main testdata/test_nbc.json.
-//
-//nolint:gosec // generated cse_cmd.sh file needs execute permissions for bootstrapping
+// This script will take in the bootstrap contract in JSON format,
+// it will be deserialized to the contract that the VHD this binary supports.
+// The contract will be used to generate cse_cmd and trigger the bootstrap process.
 func main() {
-	if len(os.Args) < parser.MinArgs {
-		log.Default().Printf("Usage: %s <input.json>", os.Args[0])
-		return
+	configFile := flag.String("bootstrap-config", "", "bootstrap configuration json filepath")
+	test := flag.Bool("test", false, "test mode")
+	flag.Parse()
+	if *configFile == "" {
+		log.Fatal("bootstrap-config is a required argument")
 	}
 
-	// Read in the JSON file
-	inputJSON, err := os.ReadFile(os.Args[1])
+	cseCmd := parseConfig(*configFile)
+	err := bootstrap(cseCmd, *test)
```

Comment:
where do we actually call the bootstrap() with test=true?

Reply:
in `linux-vhd-content-test.sh` we invoke the binary to test that it runs but don't want to actually run cse

Reply:
Could you point me where it is? 
The reason I was asking this is to understand if we really need this variable for test purpose (in the meantime it will increase chances for the parser to run in test mode unintentionally, causing provisioning failure in production), or we can simply mock a test function in the test codes.
But if it's really necessary, we still can go with it. No preference now.

Reply:
That makes sense. Thanks for clarifying.